File: e-Delivery/e-Delivery.Model/FoodItem/FoodItemGetVM.cs
using e_Delivery.Model.Category;
using e_Delivery.Model.SideDish;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace e_Delivery.Model.FoodItem
{
    public class FoodItemGetVM
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public double Price { get; set; }
        public bool IsAvailable { get; set; }
        public int LogoId { get; set; }
        public CategoriesGetVM Category { get; set; }
        public List<FoodItemPictureGetVM> FoodItemPictures { get; set; }
        public int RestaurantId { get; set; }
        public List<GetSideDishVM> SideDishes { get; set; }
        

    }
}

File: e-Delivery/UI/desktop/lib/restaurant/viewmodels/food_item_get_VM.dart
class SideDishViewModel {
  final int id;
  String name;
  double price;
  bool isAvailable;

  SideDishViewModel({
    required this.id,
    required this.name,
    required this.price,
    required this.isAvailable,
  });

  factory SideDishViewModel.fromJson(Map<String, dynamic> json) {
    return SideDishViewModel(
      id: json['id'],
      name: json['name'],
      price: json['price'].toDouble(),
      isAvailable: json['isAvailable'],
    );
  }
}

class CategoryViewModel {
  final String name;

  CategoryViewModel({required this.name});

  factory CategoryViewModel.fromJson(Map<String, dynamic> json) {
    return CategoryViewModel(name: json['name']);
  }
}

class FoodItemViewModel {
  final int id;
  final String name;
  final String description;
  final double price;
  final bool isAvailable;
  final int? logoId;
  final CategoryViewModel category;
  final List<SideDishViewModel> sideDishes;

  FoodItemViewModel({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.isAvailable,
    required this.category,
    required this.sideDishes,
    this.logoId,
  });

  factory FoodItemViewModel.fromJson(Map<String, dynamic> json) {
    return FoodItemViewModel(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: json['price'].toDouble(),
      isAvailable: json['isAvailable'],
      logoId: json['logoId'],
      category: json['category'] != null
          ? CategoryViewModel.fromJson(json['category'])
          : CategoryViewModel(name: ''),
      sideDishes: List<SideDishViewModel>.from(
          json['sideDishes'].map((x) => SideDishViewModel.fromJson(x))),
    );
  }
}
